### PR TITLE
Handle cancelDelay not being a function

### PR DESCRIPTION
**Pull request by CameronDiver**
Change-type: patch
Closes: #1080
Signed-off-by: Cameron Diver <cameron@balena.io>

### CODE REVIEW

File: src/device-state.coffee [hunk 1]
```diff
@@ -469,8 +469,11 @@ module.exports = class DeviceState extends EventEmitter
 			if !@scheduledApply? || (isFromApi && @cancelDelay)
 				@scheduledApply = { force, delay }
 				if isFromApi
-					# Cancel promise delay if call came from api to prevent waiting due to backoff
-					@cancelDelay()
+					# Cancel promise delay if call came from api to
+					# prevent waiting due to backoff (and if we've
+					# previously setup a delay)
+					if _.isFunction(@cancelDelay)
+						@cancelDelay()
```
Comment (Page-) [inline src/device-state.coffee]: @cancelDelay?()

Comment (CameronDiver) [reply src/device-state.coffee]: I'd rather make sure it was a function explicitly to be honest.

Comment (Page-) [reply src/device-state.coffee]: @cancelDelay does explicitly check it's a function and only calls it if it is, you can check the generated code to verify. If you're thinking it cutie be something other than null or a function though that should definitely be looked into as it's probably a much deeper bug

Comment (CameronDiver) [reply src/device-state.coffee]: TIL about `?()` being specifically a function check. Yeah it's certainly not ideal, but I've made big inroads into converting this file to typescript (and removing the mess that this `cancelDelay` setup is)

**Pull request merged**